src/SSVOpenHexagon/Core/main.cpp

```diff
@@ -104,7 +102,22 @@ int main(int argc, char* argv[])
 
     assets->refreshVolumes();
     window.setGameState(mg->getGame());
-    mg->init(false /* mError */);
+
+    const std::string levelName = [&]
+    {
+      std::string levelResult;
+      for(int i{0}; i < argc; ++i)
+      {
+          if(!strcmp(argv[i], "-l"))
+          {
+              levelResult = argv[++i];
+              break;
+          }
+      }
+      return levelResult;
+    }();
+
+    mg->init(false /* mError */, levelName);
```

Comment:
I would replace all of this with the `std::optional` solution above, and with the new suggested interface:

```cpp
std::optional<std::string> commandLineLevelArg;
// ...

if (commandLineLevelArg.has_value())
{
    mg->initWithLevel(false /* mError */, commandLineLevelArg.value());
}
else
{
    mg->initWithoutLevel(false /* mError */);
}
```